r-package/grf/vignettes/categorical_inputs.Rmd

```diff
@@ -0,0 +1,78 @@
+---
+title: "Categorical inputs"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{categorical_inputs}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+set.seed(123)
+```
+
+```{r setup}
+library(grf)
+library(sufrep)
+```
+
+The following example illustrates how to deal with categorical variables.
+
+Let's pretend we would like to estimate mileage per gallon (`mpg`) from number of cylinders (`cyl`),
+quarter-mile time (`qsec`), and car maker name (`maker`, created below).
+
+```{r dataset}
+# Create a categorical column with maker name
+df <- within(mtcars, {
+  # E.g. 'Mazda RX4' --> 'Mazda'
+  maker <- factor(sapply(rownames(mtcars), function(x) strsplit(x, " ")[[1]][1]))
+})
+
+x <- c("cyl", "qsec") # Continuous variables
+g <- c("maker")       # Categorical variable
+
+head(df[c(x, g)])
+```
+
+This code would raise an error, because data is not numerical.
+
+```{r problem}
+# rf <- regression_forest(X, Y)
```

Comment:
what is X and what is Y? It is evident from the context, just checking if you forgot to rename them

Reply:
Good point, thanks.